akka-persistence/src/main/scala/akka/persistence/journal/leveldb/LeveldbStore.scala

```diff
@@ -40,12 +43,30 @@ private[persistence] trait LeveldbStore extends Actor with WriteJournalBase with
 
   import Key._
 
-  def asyncWriteMessages(messages: immutable.Seq[AtomicWrite]): Future[immutable.Seq[Try[Unit]]] =
+  def asyncWriteMessages(messages: immutable.Seq[AtomicWrite]): Future[immutable.Seq[Try[Unit]]] = {
+    // collect the subscribers that are to be notified inside the future (other thread)
+    val subscribers: Map[String, Set[ActorRef]] =
+      if (!hasPersistenceIdSubscribers) Map.empty
+      else {
+        val persistenceIds = messages.foldLeft(Set.empty[String])((acc, m) ⇒ acc + m.persistenceId)
+        persistenceIds.collect {
+          case id if persistenceIdSubscribers.contains(id) ⇒
+            id -> persistenceIdSubscribers(id).toSet
+        }(collection.breakOut)
+      }
+
     Future.fromTry(Try {
-      withBatch(batch ⇒ messages.map { a ⇒
+      val result = withBatch(batch ⇒ messages.map { a ⇒
         Try(a.payload.foreach(message ⇒ addToMessageBatch(message, batch)))
       })
+      subscribers.foreach {
```

Comment:
Could call `notifyPersistenceIdChange` instead (or do you want to avoid the contains call?)